components/org.wso2.carbon.identity.discovery/src/main/java/org/wso2/carbon/identity/discovery/builders/ProviderConfigBuilder.java

```diff
@@ -140,6 +143,23 @@ public OIDProviderConfigResponse buildOIDProviderConfig(OIDProviderRequest reque
         if (OAuth2Util.getSupportedGrantTypes().contains(DEVICE_FLOW_GRANT_TYPE)) {
             providerConfig.setDeviceAuthorizationEndpoint(OAuth2Util.OAuthURL.getDeviceAuthzEPUrl());
         }
+
+        List<String> supportedTokenEndpointSigningAlgorithms = OAuthServerConfiguration.getInstance()
+                .getSupportedTokenEndpointSigningAlgorithms();
+        String[] supportedTokenEndpointSigningAlgorithmsArray = new String[supportedTokenEndpointSigningAlgorithms
+                .size()];
+        try {
+            for (int count = 0; count < supportedTokenEndpointSigningAlgorithms.size(); count++) {
+                supportedTokenEndpointSigningAlgorithmsArray[count] =  String.valueOf(OAuth2Util
+                        .mapSignatureAlgorithmForJWSAlgorithm(supportedTokenEndpointSigningAlgorithms.get(count)));
+            }
+        } catch (IdentityOAuth2Exception e) {
+            throw new ServerConfigurationException("Unsupported signature algorithm configured.", e);
+        }
+        providerConfig.setTokenEndpointAuthSigningAlgValuesSupported(supportedTokenEndpointSigningAlgorithmsArray);
+
+        providerConfig.setTlsClientCertificateBoundAccessTokensSupported(OAuthServerConfiguration.getInstance()
```

Comment:
This should be added if the relevant component is in the system only. As we don't need to have a config to add that component to the system let's not rely on a config to add this for discovery

Reply:
As the MTLS authenticator already supports it, shall we hardcode this as true without using a config? Please note that in the FAPI conformance test suite this is being checked from the well-known.

Reply:
can we check if this component is available in the system and set this to true.
If we go in that lets add that checking logic to a method inside the OAuth2Util

Reply:
In that case what we need to check is whether the MTLS authenticator is present because the logic is in there. So will add that check to a separate method in OAuth2Util.

Reply:
yes sounds good

Reply:
Fixed in ca2fd496047be88c3a8d0c0721f3b7ea411a3209